client/client.go

```diff
@@ -217,6 +217,40 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// VerifyInclusion ensures that the given leaf data has been included in the log.
+func (c *LogClient) VerifyInclusion(ctx context.Context, data []byte) error {
+	leaf := c.buildLeaf(data)
+	if err := c.UpdateRoot(ctx); err != nil {
+		return fmt.Errorf("UpdateRoot(): %v", err)
+	}
+	return c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
+}
+
+// VerifyInclusionAtIndex ensures that the given leaf data has been included in the log at a particular index.
+func (c *LogClient) VerifyInclusionAtIndex(ctx context.Context, data []byte, index int64) error {
+	leaf := c.buildLeaf(data)
+	if err := c.UpdateRoot(ctx); err != nil {
+		return fmt.Errorf("UpdateRoot(): %v", err)
+	}
+	req := &trillian.GetInclusionProofRequest{
+		LogId:     c.LogID,
+		LeafIndex: index,
+		TreeSize:  c.root.TreeSize,
+	}
+	resp, err := c.client.GetInclusionProof(ctx, req)
+	if err != nil {
+		return err
+	}
+	if len(resp.GetProof().ProofNode) < 1 {
```

Comment:
nit: I wouldn't expect the server to return 0-length proofs - that sounds like a bug to me. If you want to keep it as a defensive check that's ok, but maybe add a comment saying so?